### PR TITLE
enhance tanium threat response V2 integration

### PR DESCRIPTION
**Pull request by GuyAfik**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/41905

## Description
A few sentences describing the overall goals of the pull request's commits.

enhance the tanium threat response V2 integration & create playbooks from V1 to fit V2 as well.

## Screenshots
Paste here any images that will help the reviewer

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml [hunk 1]
```diff
@@ -451,21 +467,63 @@ script:
...
       secret: false
+    - name: status
+      description: "Get the connections that match only to a specific status. possible\
+        \ states: disconnected, timeout, waiting, connected. Can be a list, for example\
+        \ status=timeout,waiting"
+      required: false
+      isArray: true
+      defaultValue: ""
+      predefined:
+      - ""
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml:494]: remove those empty `- ""`

**Review by Itay4 — CHANGES_REQUESTED** [review]

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md [hunk 2]
```diff
@@ -638,52 +643,10 @@ Returns all connections.
 
 
 #### Command Example
-```!tanium-tr-list-connections```
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md:641]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml [hunk 3]
```diff
@@ -1,11 +1,11 @@
-id: Tanium Threat Response - Create Connection
+id: Tanium Threat Response V1 - Create Connection
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml:1]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md:641]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml:1]: ack

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
 version: -1
 fromversion: "5.0.0"
 contentitemfields:
   propagationLabels:
   - all
-name: Tanium Threat Response - Create Connection
-description: Creates a connection to a remote destination from Tanium.
+name: Tanium Threat Response V1 - Create Connection
+description: Creates a connection to a remote destination from Tanium (V1 only).
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Create_Connection.yml:8]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Request_File_Download.yml [hunk 5]
```diff
@@ -1,41 +1,35 @@
 elasticcommonfields: {}
-id: Tanium Threat Response - Request File Download
+id: Tanium Threat Response V1 - Request File Download
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV1_-_Request_File_Download.yml:2]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 6]
```diff
@@ -0,0 +1,200 @@
+description: Creates a connection to a remote destination from Tanium (V2 only).
+id: Tanium Threat Response V2 - Create Connection
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:2]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 7]
```diff
@@ -0,0 +1,200 @@
+description: Creates a connection to a remote destination from Tanium (V2 only).
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:1]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+  required: true
+  value:
+    simple: ""
+- description: ""
+  key: platform
+  playbookInputQuery:
+  required: true
+  value:
+    simple: Linux
+name: Tanium Threat Response V2 - Create Connection
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:28]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 9]
```diff
@@ -0,0 +1,200 @@
+description: Creates a connection to a remote destination from Tanium (V2 only).
+id: Tanium Threat Response V2 - Create Connection
+inputs:
+- description: ""
+  key: client-id
+  playbookInputQuery:
+  required: true
+  value:
+    simple: "2232836718"
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:9]: a mistake, removed them

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml [hunk 10]
```diff
@@ -0,0 +1,452 @@
+description: Request file download from Tanium (V2 only).
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml:1]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 11]
```diff
@@ -1017,38 +1036,63 @@ def delete_local_snapshot(client, data_args) -> Tuple[str, dict, Union[list, dic
...
+        client (Client): client that connects to the Tanium-Threat-Response API.
+        command_args (dict): command arguments entered by the user. (limit, offset, ip, status, hostname, platform).
 
+    Returns:
+        tuple (str, dict, list[dict]): table output, context output and raw response by the Tanium-Threat-Response API.
     """
-    limit = arg_to_number(data_args.get('limit'))
-    offset = arg_to_number(data_args.get('offset'))
-    raw_response = client.do_request('GET', '/plugin/products/threat-response/api/v1/conns')
+    _ip, _status, _hostname, _platform, _id = "ip", "status", "hostname", "platform", "id"
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:1053]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 12]
```diff
@@ -1017,38 +1036,63 @@ def delete_local_snapshot(client, data_args) -> Tuple[str, dict, Union[list, dic
...
+
+    limit = arg_to_number(command_args.get('limit'))
+    offset = arg_to_number(command_args.get('offset'))
+    ids = argToList(arg=command_args.get("id"))
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:1057]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 13]
```diff
@@ -167,6 +167,25 @@ def convert_to_int(int_to_parse: Any) -> Optional[int]:
     return res
 
 
+def are_filters_match_response_content(all_filter_arguments: list, api_response: dict) -> bool:
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:170]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 14]
```diff
@@ -167,6 +167,25 @@ def convert_to_int(int_to_parse: Any) -> Optional[int]:
...
+        all_filter_arguments (list[tuple]): pairs of filter arguments inputs & a response key.
+        api_response (dict): api response.
+
+    Returns:
+        bool: True if in any of the filter arguments there was a match, False otherwise.
+    """
+    for arguments in all_filter_arguments:
+        command_args, resp_key = arguments
+        for arg in command_args:
+            if arg == api_response[resp_key]:
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:184]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 15]
```diff
@@ -1790,20 +1872,39 @@ def get_task_by_id(client, data_args) -> Tuple[str, dict, Union[list, dict]]:
...
+        tuple (str, dict, list[dict]): table output, context output and raw response by the Tanium-Threat-Response API.
     """
-    limit = arg_to_number(data_args.get('limit'))
-    offset = arg_to_number(data_args.get('offset'))
+    _status = "status"
+
+    limit = arg_to_number(command_args.get('limit'))
+    offset = arg_to_number(command_args.get('offset'))
+    statuses = argToList(arg=command_args.get(_status))
+    hostnames = argToList(arg=command_args.get("hostname"))
```
[the remark this answers is not in the document]
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:1895]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml (hunk 1 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml:494]: done

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md [hunk 16]
```diff
@@ -616,6 +616,11 @@ Returns all connections.
 | --- | --- | --- |
 | limit | The maximum number of connections to return. Default is 50. | Optional | 
 | offset | The offset number to begin listing connections. Default is 0. | Optional | 
+| status | Get the connections that match only to a specific status. possible states: disconnected, timeout, waiting, connected. Can be a list, for example status=timeout,waiting. Possible values are: . | Optional | 
+| ip | Get the connections that match only to a specific IP. Can be a list, for example ip=1.1.1.1,8.8.8.8. Possible values are: . | Optional | 
+| platform | Get the connections that match only to a specific platform (OS). (Windows, Linux). Can be a list, for example platform=Windows,Linux . Possible values are: . | Optional |
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md:621]: why do we have possible values in all descriptions here?

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 17]
```diff
@@ -0,0 +1,200 @@
...
+        "width": 380,
+        "x": 50,
+        "y": 60
+      }
+    }
+  }
+tests:
+- Tanium Threat Response V2 Test
+- Tanium Threat Response Test
+fromversion: 5.0.0
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:200]:
```suggestion
fromversion: 6.0.0
```

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml [hunk 18]
```diff
@@ -0,0 +1,200 @@
...
+        "height": 550,
+        "width": 380,
+        "x": 50,
+        "y": 60
+      }
+    }
+  }
+tests:
+- Tanium Threat Response V2 Test
+- Tanium Threat Response Test
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:199]: why do we have both tests here?

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml [hunk 19]
```diff
@@ -0,0 +1,452 @@
...
+    "paper": {
+      "dimensions": {
+        "height": 1610,
+        "width": 492.5,
+        "x": 50,
+        "y": 50
+      }
+    }
+  }
+fromversion: 5.0.0
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml:449]:
```suggestion
fromversion: 6.0.0
```

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml [hunk 20]
```diff
@@ -0,0 +1,452 @@
...
+tests:
+- Tanium Threat Response V2 Test
+- Tanium Threat Response Test
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml:452]: why do we have both tests here?

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py [hunk 21]
```diff
@@ -1612,41 +1679,58 @@ def get_process_tree(client, data_args) -> Tuple[str, dict, Union[list, dict]]:
...
+        client (Clinet): client that connects to the Tanium-Threat-Response API.
+        command_args (dict): command arguments entered by the user. (limit, offset, hostname, sort, type).
 
+    Returns:
+        tuple (str, dict, list[dict]): table output, context output and raw response by the Tanium-Threat-Response API.
     """
-    limit = arg_to_number(data_args.get('limit'))
-    offset = arg_to_number(data_args.get('offset'))
-    sort = data_args.get('sort')
+    _limit, _offset, _hostname, _sort, _type, _created_at = "limit", "offset", "hostname", "sort", "type", "createdAt"
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.py:1696]: same comment here about the code standard

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml [hunk 22]
```diff
@@ -4,2191 +4,2351 @@ commonfields:
   id: Tanium Threat Response v2
   version: -1
 configuration:
-- display: Hostname, IP address, or server URL
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml:7]: lets avoid the big diff here, and apply here only the relevant changes

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md [hunk 23]
```diff
@@ -0,0 +1,17 @@
...
+#### Integrations
+##### Tanium Threat Response v2
+- Added the ability to filter command '**tanium-tr-list-connections**' response according to status/ip/platform/hostname/id.
+- Added the ability to filter command '**tanium-tr-get-system-status**' response according to status/ip_client/ip_server/port/hostname.
+- Added the ability to filter command '**tanium-tr-event-evidence-list**' response according to hostname/type.
+- Added the ability to filter command '**tanium-tr-list-file-downloads**' response according to hash/hostname/process_time_start.
+
+#### Playbooks
+#### Tanium Threat Response - Create Connection
+- - Update playbook **playbook-Tanium Threat Response - Create Connection** description
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:11]:
```suggestion
- General documentation improvements.
```

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md [hunk 24]
```diff
@@ -0,0 +1,17 @@
...
+#### Tanium Threat Response - Request File Download
+- Update playbook **playbook-Tanium Threat Response - Request File Download** description
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:13]:
```suggestion
- General documentation improvements.
```

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md [hunk 25]
```diff
@@ -0,0 +1,17 @@
...
+#### Tanium Threat Response - Create Connection v2
+- Add playbook **playbook-Tanium Threat Response V2 - Create Connection** to correspond TaniumThreatResponseV2 Integration.
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:15]: put here the playbook desc
also, follow the `New` entity release notes convention

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md [hunk 26]
```diff
@@ -0,0 +1,17 @@
...
+##### Tanium Threat Response - Request File Download v2
+- Add playbook **playbook-Tanium Threat Response V2 - Request File Download** to correspond TaniumThreatResponseV2 Integration.
```
Comment (Itay4) [inline Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:17]: put here the playbook desc
also, follow the `New` entity release notes convention

**Review by Itay4 — COMMENTED** [review]
please add test playbooks and docs for the new playbooks you added

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md (hunk 16 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/README.md:621]: not sure why the plugin added those, removed them

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml (hunk 17 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:200]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml (hunk 18 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Create_Connection.yml:199]: not sure, changed to the playbook that should test this playbook

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml (hunk 19 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml:449]: done

File: Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml (hunk 20 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Playbooks/playbook-Tanium_Threat_ResponseV2_-_Request_File_Download.yml:452]: not sure, changed to the playbook that should test this playbook

File: Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml (hunk 22 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/Integrations/TaniumThreatResponseV2/TaniumThreatResponseV2.yml:7]: done, its a consequence of the plugin, so i just did it manually instead

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 23 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:11]: done

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 25 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:15]: not sure where i can find those release notes conventions? in developer docs?

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 26 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:17]: not sure where i can find those release notes conventions? in developer docs?

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 23 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:11]: > please add test playbooks and docs for the new playbooks you added

done

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 25 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:15]: I managed to do that simply by re-running the generate docs of the sdk

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md (hunk 26 above)
Comment (GuyAfik) [reply Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:17]: I managed to do that simply by re-running the generate docs of the sdk

File: Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md [hunk 27]
```diff
@@ -1,3 +1,3 @@
 #### Integrations
 ##### Tanium Threat Response
-- Updated the Docker image to: *demisto/python3:3.9.8.24399*.
+- Updated the Docker image to: *demisto/python3:3.9.8.24399*.
```
Comment (GuyAfik) [inline Packs/TaniumThreatResponse/ReleaseNotes/2_0_3.md:3]: had to rebase it against the master as the docker image was updated, not sure why git recognizes it as a change as it is basically the same line

**Review by Itay4 — APPROVED** [review]
looks good 👍

**Pull request merged**